### PR TITLE
Removes References To Built-in Custom Data Implementations

### PR DESCRIPTION
**Pull request by gsalaz98**
#### Description

This PR removes references to custom data types that have been added directly into the Lean repo over the course of the last two years.

Some new "Iconic" custom data types were added to keep some of the important Lean tests related to custom data. These data types are mocked versions of actual data types, containing functionality that was previously present in certain data types.

The new recommended pattern of adding data types to be supported in the cloud is via the instructions contained within the https://github.com/QuantConnect/Lean.DataSource.SDK repo

Data types removed & their new repos:
  * https://github.com/QuantConnect/Lean.DataSource.SEC (`SECReport8K, SECReport10K, SECReport10Q`)
  * https://github.com/QuantConnect/Lean.DataSource.TradingEconomics (`TradingEconomicsCalendar, TradingEconomicsIndicator`)
  * https://github.com/QuantConnect/Lean.DataSource.Estimize (`EstimizeConsensus, EstimizeRelease, EstimizeEstimate`)
  * https://github.com/QuantConnect/Lean.DataSource.SmartInsiderIntentionsTransactions (`SmartInsiderIntention, SmartInsiderTransaction`)
  * https://github.com/QuantConnect/Lean.DataSource.BenzingaNews (`BenzingaNews`)
  * https://github.com/QuantConnect/Lean.DataSource.USTreasury (`USTreasuryYieldCurveRate`)
  * https://github.com/QuantConnect/Lean.DataSource.USEnergy (`USEnergy`)
  * https://github.com/QuantConnect/Lean.DataSource.FRED (`Fred`)
  * https://github.com/QuantConnect/Lean.DataSource.TiingoNews (`TiingoNews`)
  * https://github.com/QuantConnect/Lean.DataSource.CBOE (`CBOE`)
  * https://github.com/QuantConnect/Lean.DataSource.QuiverQuantWikipedia (`QuiverWikipedia`)
  * https://github.com/QuantConnect/Lean.DataSource.QuiverQuantWallStreetBets (`QuiverWallStreetBets`)
  * https://github.com/QuantConnect/Lean.DataSource.QuiverQuantEventsBeta (`QuiverEventsBeta`)
  * https://github.com/QuantConnect/Lean.DataSource.QuiverQuantCongressTrading (`QuiverCongress`)
  * https://github.com/QuantConnect/Lean.DataSource.VIXCentral (`VIXCentral`)

:warning: This is a work in progress, since there's some more references we have to remove in other repos, so please don't merge yet.
#### Related Issue

N/A
#### Motivation and Context

Removes class name clashes and unifies implementations of the same data types going forward, without the need to update this repo; Allows data vendors to implement their own data types without having to merge directly into this repo.
#### Requires Documentation Change

Yes.

Instead of importing from the `QuantConnect.Data.Custom.*` namespaces, we should direct users to import from `QuantConnect.DataSource.*`
#### How Has This Been Tested?

All unit tests passing, cloud has been working for a few weeks with new data types when not importing the data from the Custom.* namespace
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [ ] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: ToolBox/Program.cs [hunk 1]
```diff
@@ -131,50 +125,6 @@ public static void Main(string[] args)
                     case "binancedownloader":
                         BinanceDownloaderProgram.DataDownloader(tickers, resolution, fromDate, toDate);
                         break;
-                    case "secdl":
-                    case "secdownloader":
```
Comment (Martin-Molinero) [inline ToolBox/Program.cs:135]: Need to update the help print statement above too? 👍

File: Tests/Common/ProtobufSerializationTests.cs [hunk 2]
```diff
@@ -34,117 +31,6 @@ namespace QuantConnect.Tests.Common
     [TestFixture, Parallelizable(ParallelScope.All)]
     public class ProtobufSerializationTests
     {
-        [Test]
-        public void SerializeRoundTripEstimizeRelease()
```
Comment (Martin-Molinero) [inline Tests/Common/ProtobufSerializationTests.cs:38]: Think we should add the Iconic data types round trip serialization as a test of serializing these custom data types 👍

File: ToolBox/Program.cs (hunk 1 above)
Comment (Martin-Molinero) [reply ToolBox/Program.cs:135]: Ah never mind u already did 😄

**Comment by gsalaz98** [comment]
Re-creating the PR with branch living in this repo instead of the one on my fork

**Pull request closed without merging**